src/ngraph/frontend/onnx_import/op/reduce.cpp

```diff
@@ -32,31 +32,67 @@ namespace ngraph
             {
                 NodeVector reduce_mean(const Node& node)
                 {
-                    auto input_shape = node.get_ng_inputs().at(0)->get_shape();
-                    auto reduction_axes = reduction::detail::get_reduction_axes(node);
-                    std::size_t elem_count_product =
-                        std::accumulate(std::begin(reduction_axes),
-                                        std::end(reduction_axes),
-                                        1UL,
-                                        [&input_shape](const std::size_t& a, const std::size_t& b) {
-                                            return a * input_shape.at(b);
-                                        });
-
-                    auto sum_node = std::shared_ptr<ngraph::Node>{reduction::make_ng_reduction_op(
+                    const auto data = node.get_ng_inputs().at(0);
+                    const auto& data_ps = data->get_output_partial_shape(0);
+
+                    // sum up the input data along the reduction axes
+                    const auto sum_node = reduction::make_ng_reduction_op(
                         node,
-                        node.get_ng_inputs().at(0),
+                        data,
                         std::make_shared<default_opset::ReduceSum,
                                          const std::shared_ptr<ngraph::Node>&,
                                          const std::shared_ptr<ngraph::Node>&,
-                                         bool>)};
+                                         bool>);
+
+                    if (data_ps.is_static())
+                    {
+                        const auto reduction_axes = reduction::detail::get_reduction_axes(node);
+                        const auto input_shape = data_ps.to_shape();
+
+                        // calculate the product of dimensions pointed to by reduction axes
+                        // this value represents the number of input tensor values that were reduced
+                        size_t elem_count_product = 1U;
+                        for (const auto axis : reduction_axes)
+                        {
+                            elem_count_product *= input_shape.at(axis);
+                        }
+
+                        auto const_node = default_opset::Constant::create(
+                            sum_node->get_element_type(),
+                            sum_node->get_shape(),
+                            std::vector<std::size_t>(shape_size(sum_node->get_shape()),
```

Comment:
You should be able to leave this const a scalar, omitting the shape, allowing `Divide` to broadcast as needed.